parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -134,6 +134,28 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "migPartitionScript"}}
 
+{{if IsMariner}}
```

Comment:
so is the plan to distribute this unit/timer to older VHDs currently in the wild?

adding customdata payloads here will have it such that certain operations (such as SP reset) will cause VMs running with older node image versions (and intended customdata/cse) to receive these payloads when the VMs are reimaged - is that something we want?

generally if we add something directly do the VHD like you're doing with these payloads, we don't put into CSE/customdata.

Reply:
The requirement is to have the unit/timer backfilling into old clusters. The final goal is to rollout this feature to all clusters, including existing/new clusters.

So yes. This is what I want.

Reply:
@cameronmeissner , This systemd unit is triggered by an internal RP API. So with this systemd unit alone, there will be noop on the nodes. Our plan is to use vm extension(npd) to push these units to older images.  Before this is done, we will only trigger the RP API when the target agent pool with specific node image versions (> the next release version). So even the old images gets these units with a reimage, there will be noop on the older images nodes. We will only remove the filtering when the systemd unit is available everywhere.